cdap-ui/app/cdap/services/helpers.js

```diff
@@ -489,6 +489,37 @@ function connectWithStore(store, WrappedComponent, ...args) {
     return <ConnectedWrappedComponent {...props} store={store} />;
   };
 }
+/**
+ * This function formats the graphQl errors by error type
+ * { errorType: ['message1', 'message2'] ....} will be
+ * the format of categorized errors.
+ */
+function categorizeGraphQlErrors(error) {
+  const GENERIC_ERROR_ORIGIN = 'generic';
+  const graphQLErrors = objectQuery(error, 'graphQLErrors') || [];
+  const networkErrors = objectQuery(error, 'networkError') || [];
+  const errorsByOrigin = {};
+  if (graphQLErrors.length === 0 && networkErrors.length === 0 && error) {
+    errorsByOrigin[GENERIC_ERROR_ORIGIN] = error.message;
+  }
+  graphQLErrors.forEach(error => {
+    const errorOrigin = objectQuery(error, 'extensions', 'exception', 'errorOrigin') || GENERIC_ERROR_ORIGIN;
+    if (errorsByOrigin.hasOwnProperty(errorOrigin)) {
+      errorsByOrigin[errorOrigin] = [error.message];
+    }
+    else {
+      errorsByOrigin[errorOrigin].push([error.message]);
+    }
+  });
+  networkErrors.forEach(error => {
+    if (errorsByOrigin.hasOwnProperty('network')) {
+      errorsByOrigin['network'] = [error.message];
+    } else {
+      errorsByOrigin['network'].push([error.message]);
```

Comment:
Is the condition above correct? If I'm reading the condition correctly, `errorsByOrigin['network']` will always be `undefined` here.

Reply:
Fixed the condition